visualization.py

```
@@ -0,0 +1,57 @@
+        decoded_supplier = decode_answer(row['mg_id'])  
+        decoded_buyer = decode_answer(row['buyer'])  
+        if decoded_buyer not in G.nodes:
+            G.add_node(decoded_buyer, country=row['buyer_country'], is_subsid=row['buyer_is_subsid'])
+        if decoded_supplier not in G.nodes:
+            G.add_node(decoded_supplier)
+        G.add_edge(decoded_buyer, decoded_supplier, weight=row['amount']/1000.00, date=row['date'])
+    return G
+
+def create_supplier_graph(supplier_data: pd.DataFrame):
+    G = nx.Graph()
+    for index, row in supplier_data.iterrows():
```

Ruff (0.8.2)

Loop control variable `index` not used within loop body

Rename unused `index` to `_index`

(B007)